consensus/replay.go

```diff
@@ -299,6 +355,19 @@ func (h *Handshaker) ReplayBlocks(
 		"stateHeight",
 		stateBlockHeight)
 
+	// Check restorable conditions for localsync
+	if storeBlockHeight == 0 && appBlockHeight > 0 {
+		var syncErr error
+		if state, _, syncErr = h.localSync(uint64(appBlockHeight)); syncErr != nil {
+			panic("Unable to use RPC to update state")
+		}
+		h.logger.Info("localSync resulting state.Version.Consensus.Block", "state", state.Version.Consensus.Block)
+
+		// Assume Heights Restored, Update Heights for edge cases and constraints on the storeBlockHeight and storeBlockBase.
+		storeBlockHeight = appBlockHeight
```

Comment:
I would say that these values should be read from the stores, not blindly updated here.